Yoast/wordpress-seo
# P3-589 Keep original frontpage Open Graph settings for downgrades

## Context


* When upgrading from 16.2 to 16.3 the Open Graph frontpage settings should be moved to new fields, but the old values should be kept in the old location so downgrades from 16.3 to 16.2 still show the previous values for the frontpage settings.

## Summary


This PR can be summarized in the following changelog entry:

* Fixes a bug where the frontpage settions would be lost when downgrading to an earlier version of the plugin.

## Relevant technical choices:

* I added the new fields to the tracking settings and also kept the old ones in there.

## Test instructions

### Test instructions for the acceptance test before the PR gets merged
This PR can be acceptance tested by following these steps:

**Testing option migration**

* Reset trunk at an older version of the plugin (`$ git checkout trunk && git reset --hard tags/16.2`)
* Build the plugin as usual (`$ composer install && yarn && grunt build`)
* Ensure you have custom frontpage title, description and image set under SEO -> Social -> Facebook -> Frontpage settings.
* Save your changes.
* Checkout this branch and build accordingly (`$ composer install && yarn && grunt build`).
* Use the test helper to change the current version back to 16.2.
* Navigate to SEO -> Social -> Facebook.
* See that the Frontpage settings are no longer there and instead you see a notification pointing you to the Search Appearance section.
* Inspect the DOM of the page and find the 4 hidden elements below that notification.
* See that the image, title and description that you set before, are reflected in those hidden elements.
* Click in the link in the notification. You will be sent to the new location for the frontpage settings.
* See that the image, title and description that you set before, is reflected in the new location.
* Reset trunk at an older version of the plugin (`$ git checkout trunk && git reset --hard tags/16.2`)
* Build the plugin as usual (`$ composer install && yarn && grunt build`)
* Check that SEO -> Social -> Facebook -> Frontpage now shows your initial data again

**Testing the form**

Since the field names have changed, it's important to test the form again.

* Checkout this branch and build accordingly (`$ composer install && yarn && grunt build`).
* Navigate to SEO -> Search Appearance -> Content Types > Frontpage.
* Edit the image, title and description.
* Save your changes.
* Navigate to the homepage of your environment and inspect the source.
* See that the changes you made to the title and description are reflected in the `og:title`, `og:description` and `og:image` meta fields.

### Test instructions for QA when the code is in the RC




QA can test this PR by following these steps:

**Testing option migration**

* Install an older version of the plugin
* Ensure you have custom frontpage title, description and image set under SEO -> Social -> Facebook -> Frontpage settings.
* Save your changes.
* Upgrade to the new version of the plugin.
* Navigate to SEO -> Social -> Facebook.
* See that the Frontpage settings are no longer there and instead you see a notification pointing you to the Search Appearance section.
* Inspect the DOM of the page and find the 4 hidden elements below that notification.
* See that the image, title and description that you set before, are reflected in those hidden elements.
* Click in the link in the notification. You will be sent to the new location for the frontpage settings.
* See that the image, title and description that you set before, is reflected in the new location.
* Replace the plugin with the older version again.
* Check that SEO -> Social -> Facebook -> Frontpage now shows your initial data again.

**Testing the form**

Since the field names have changed, it's important to test the form again.

* Install the plugin.
* Navigate to SEO -> Search Appearance -> Content Types > Frontpage.
* Edit the image, title and description.
* Save your changes.
* Navigate to the homepage of your environment and inspect the source.
* See that the changes you made to the title and description are reflected in the `og:title`, `og:description` and `og:image` meta fields.

## Impact check

This PR affects the following parts of the plugin, which may require extra testing:

*

## UI changes


## Documentation


## Quality assurance


Fixes #
